File: WaterMango/ClientApp/src/app/_models/plantModel.ts
import { WateringHistory } from './wateringHistory';
export class PlantModel {
  constructor() {
    this.wateringHistory = [];
    this.isWatering = false;
  }
  token: string;
  name: string;
  location: string;
  wateringHistory: WateringHistory[];
  isWatering: boolean;
  imageUrl: string;
}

File: WaterMango/Model/PlantModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WaterMango.Model
{
    public class PlantModel
    {
        [JsonProperty("token")]
        public Guid token { get; set; }
        [JsonProperty("name")]
        public string name { get; set; }
        [JsonProperty("location")]
        public string location { get; set; }
        [JsonProperty("wateringHistory")]
        public List<WateringHistoryModel> wateringHistory { get; set; }
        [JsonProperty("isWatering")]
        public bool isWatering { get; set; }
        [JsonProperty("imageUrl")]
        public string imageUrl { get; set; }
    }
}
